src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -23,3 +28,67 @@ export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
 
   return Uint8Array.from(colors.flat());
 }
+
+export function usePanZoom(camera: OrthographicCamera): ReactThreeFiber.Events {
+  const [startOffsetPosition, setStartOffsetPosition] = useState<Vector3>();
+
+  const onPointerDown = useCallback(
+    (evt: PointerEvent) => {
+      evt.stopPropagation();
+
+      const { currentTarget, pointerId } = evt as React.PointerEvent;
+      currentTarget.setPointerCapture(pointerId);
+
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+      setStartOffsetPosition(camera.position.clone().add(projectedPoint));
+    },
+    [camera]
+  );
+
+  const onPointerUp = useCallback((evt: PointerEvent) => {
+    evt.stopPropagation();
+
+    const { currentTarget, pointerId } = evt as React.PointerEvent;
+    currentTarget.releasePointerCapture(pointerId);
```

Comment:
Pointer capturing allows pointer events to remain targeted to the initial target object (i.e. the mesh) throughout the drag, event when the pointer moves outside of the bounds of the mesh (e.g. when it moves onto the explorer sidebar).